src/wp-includes/class-wp-scripts.php

```diff
@@ -464,7 +519,19 @@ public function print_inline_script( $handle, $position = 'after', $display = tr
 		$output = trim( implode( "\n", $output ), "\n" );
 
 		if ( $display ) {
-			printf( "<script%s id='%s-js-%s'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), esc_attr( $position ), $output );
+			if ( 'after-non-standalone' === $position ) {
```

Comment:
See my other related comments about decoupling the position from whether inline scripts are standalone.

Instead of this condition here, the condition should then be:
* If the handle's main script tag uses either `defer` or `async`...
* ... _and_ if the `$position` is "after"...
* ... _and_ if `$this->get_data( 'after-is-standalone' )` returns `false`

Only then the inline script should be wrapped with this alternative markup to not immediately execute as a script.